### ISSUE TITLE
Rstudio crashes on data.table::fread with verbose=TRUE

### ISSUE THREAD

**Issue by chrisvwn** [rstudio/rstudio#13130]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2023.03.1 Build 446
    OS Version      : Ubuntu 20.04
    R Version       : 4.2.3

### Steps to reproduce the problem
Attempt to read a space-delimited file using data.table::fread with verbose=TRUE. data.table version 1.14.8.

### Describe the problem in detail
When reading in a file, in this case a space-delimited log file, with data.table::fread, setting verbose=TRUE causes rstudio to crash with errors. The full verbose output of the command is below. Removing the verbose=TRUE option allows the operation to complete successfully. Also, this does not seem to affect CSVs. Attached is a small sample file 
[sweep_20230513_1100.log](https://github.com/rstudio/rstudio/files/11485015/sweep_20230513_1100.log). Rstudio crash log also attached 
[diagnostics-report.txt](https://github.com/rstudio/rstudio/files/11485085/diagnostics-report.txt)

> R version 4.2.3 (2023-03-15) -- "Shortstop Beagle"
> Copyright (C) 2023 The R Foundation for Statistical Computing
> Platform: x86_64-pc-linux-gnu (64-bit)
> 
> R is free software and comes with ABSOLUTELY NO WARRANTY.
> You are welcome to redistribute it under certain conditions.
> Type 'license()' or 'licence()' for distribution details.
> 
>   Natural language support but running in an English locale
> 
> R is a collaborative project with many contributors.
> Type 'contributors()' for more information and
> 'citation()' on how to cite R or R packages in publications.
> 
> Type 'demo()' for some demos, 'help()' for on-line help, or
> 'help.start()' for an HTML browser interface to help.
> Type 'q()' to quit R.
> 
> > library(dplyr)
> 
> Attaching package: ‘dplyr’
> 
> The following objects are masked from ‘package:stats’:
> 
>     filter, lag
> 
> The following objects are masked from ‘package:base’:
> 
>     intersect, setdiff, setequal, union
> 
> > library(data.table)
> data.table 1.14.8 using 4 threads (see ?getDTthreads).  Latest news: r-datatable.com
>
> Attaching package: ‘data.table’
>
> The following objects are masked from ‘package:dplyr’:
>
>    between, first, last
>
> >   nrow_read <- 1200000
> >   nrow_skip <- 0
> >   nrow_df <- nrow_read
> >     swp_in <- data.table::fread(input = "xxxxxxx.log", 
> +                                 header = F,
> +                                 integer64 = "numeric",
> +                                 skip = nrow_skip,
> +                                 nrow = nrow_read,
> +                                 verbose = TRUE)
>   OpenMP version (_OPENMP)       201511
>   omp_get_num_procs()            8
>   R_DATATABLE_NUM_PROCS_PERCENT  unset (default 50)
>   R_DATATABLE_NUM_THREADS        unset
>   R_DATATABLE_THROTTLE           unset (default 1024)
>   omp_get_thread_limit()         2147483647
>   omp_get_max_threads()          8
>   OMP_THREAD_LIMIT               unset
>   OMP_NUM_THREADS                unset
>   RestoreAfterFork               true
>   data.table is using 4 threads with throttle==1024. See ?setDTthreads.
> Input contains no \n. Taking this to be a filename to open
> [01] Check arguments
>   Using 4 threads (omp_get_max_threads()=8, nth=4)
>   NAstrings = [<<NA>>]
>   None of the NAstrings look like numbers.
>   skip num lines = 0
>   show progress = 1
>   0/1 column will be read as integer
> [02] Opening the file
>   Opening file xxxxxxx.log
>   File opened, size = 24.88MB (26085531 bytes).
>   Memory mapped ok
> [03] Detect and skip BOM
> [04] Arrange mmap to be \0 terminated
>   \n has been found in the input and different lines can end with different line endings (e.g. mixed \n and \r\n in one file). This is common and ideal.
> [05] Skipping initial rows if needed
>   Positioned on line 1 starting: <<2023-05-13, 16:57:31.609397, 0>>
> [06] Detect separator, quoting rule, and ncolumns
>   Detecting sep automatically ...
>   sep=','  with 100 lines of 11 fields using quote rule 0
>   Detected 11 columns on line 1. This line is either column names or first data row. Line starts as: <<2023-05-13, 16:57:31.609397, 0>>
>   Quote rule picked = 0
>   fill=false and the most number of columns found is 11
> [07] Detect column types, good nrow estimate and whether first row is column names
>   'header' changed by user from 'auto' to false
>   Number of sampling jump points = 100 because nrow limit (1200000) supplied
>   Type codes (jump 000)    : AC557577777  Quote rule 0
>   All rows were sampled since file is small so we know nrow=100 exactly
> [08] Assign column names
> [09] Apply user overrides on column types
>   After 0 type and 0 drop user overrides : AC557577777
> [10] Allocate memory for the datatable
>   Allocating 11 column slots (11 - 0 dropped) with 100 rows
> [11] Read the data
>   jumps=[0..1), chunk_size=1048576, total_size=26085530
> [1774405:1774406:20230516,094403.900317:ERROR file_io_posix.cc:148] open /home/xxxx/.r/crashpad_database/pending/873257ac-6852-4f6f-918f-5b6912c86b20.lock: File exists (17)
> [1774405:1774405:20230516,094403.903690:ERROR process_memory_range.cc:86] read out of range
> [1774405:1774405:20230516,094403.903710:ERROR elf_image_reader.cc:558] missing nul-terminator
> [1774405:1774405:20230516,094403.903865:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.907290:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.907368:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.907447:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.907526:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.907591:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.907659:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.908195:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.908261:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.908879:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.908970:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.910021:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.910095:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.910161:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.910229:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.910291:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.910356:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.910420:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.910484:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774405:20230516,094403.910554:ERROR elf_dynamic_array_reader.h:61] tag not found
> [1774405:1774406:20230516,094403.933003:ERROR directory_reader_posix.cc:42] opendir: No such file or directory (2)

### Describe the behavior you expected
Command should read the file successfully providing verbose output and return the contents of the file just as if verbose=TRUE was not supplied as a parameter.
[diagnostics-report.txt](https://github.com/rstudio/rstudio/files/11484940/diagnostics-report.txt)

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by jonvanausdeln** [comment]
@chrisvwn , thank you for the report.  Unfortunately, I'm not able to reproduce your issue.  Does this happen with any size of space delimited file?

**Comment by chrisvwn** [comment]
@jonvanausdeln Yes, it seemed to happen on anything from 20MB up. Let me try again to see if I can find any other distinguishing factors.

**Comment by jonvanausdeln** [comment]
@chrisvwn , thanks for the additional information.  I was trying to reproduce with a small example file.
Also, have you tried on a standalone R command shell/line ?  Outside of RStudio. Just wondering if works outside RStudio

**Comment by kevinushey** [comment]
Seems like this may be a bug in `data.table`. I was able to reproduce a crash when using a version of R built with sanitizers:

```
> library(data.table)
data.table 1.14.8 using 5 threads (see ?getDTthreads).  Latest news: r-datatable.com
nrow_read <- 1200000
nrow_skip <- 0
nrow_df <- nrow_read
swp_in <- data.table::fread(
  input = "~/Downloads/sweep_20230513_1100.log",
  header = F,
  integer64 = "numeric",
  skip = nrow_skip,
  nrow = nrow_read,
  verbose = TRUE
)
> nrow_read <- 1200000
> nrow_skip <- 0
> nrow_df <- nrow_read
> swp_in <- data.table::fread(
+   input = "~/Downloads/sweep_20230513_1100.log",
+   header = F,
+   integer64 = "numeric",
+   skip = nrow_skip,
+   nrow = nrow_read,
+   verbose = TRUE
+ )
  OpenMP version (_OPENMP)       201811
  omp_get_num_procs()            10
  R_DATATABLE_NUM_PROCS_PERCENT  unset (default 50)
  R_DATATABLE_NUM_THREADS        unset
  R_DATATABLE_THROTTLE           unset (default 1024)
  omp_get_thread_limit()         2147483647
  omp_get_max_threads()          10
  OMP_THREAD_LIMIT               unset
  OMP_NUM_THREADS                unset
  RestoreAfterFork               true
  data.table is using 5 threads with throttle==1024. See ?setDTthreads.
Input contains no \n. Taking this to be a filename to open
[01] Check arguments
  Using 5 threads (omp_get_max_threads()=10, nth=5)
  NAstrings = [<<NA>>]
  None of the NAstrings look like numbers.
  skip num lines = 0
  show progress = 1
  0/1 column will be read as integer
[02] Opening the file
  Opening file /Users/kevin/Downloads/sweep_20230513_1100.log
  File opened, size = 1.478MB (1549843 bytes).
  Memory mapped ok
[03] Detect and skip BOM
[04] Arrange mmap to be \0 terminated
  \n has been found in the input and different lines can end with different line endings (e.g. mixed \n and \r\n in one file). This is common and ideal.
[05] Skipping initial rows if needed
  Positioned on line 1 starting: <<2023-05-13, 11:59:51.773327, 0>>
[06] Detect separator, quoting rule, and ncolumns
  Detecting sep automatically ...
  sep=','  with 100 lines of 11 fields using quote rule 0
  Detected 11 columns on line 1. This line is either column names or first data row. Line starts as: <<2023-05-13, 11:59:51.773327, 0>>
  Quote rule picked = 0
  fill=false and the most number of columns found is 11
[07] Detect column types, good nrow estimate and whether first row is column names
  'header' changed by user from 'auto' to false
  Number of sampling jump points = 10 because nrow limit (1200000) supplied
  Type codes (jump 000)    : AC557577777  Quote rule 0
  All rows were sampled since file is small so we know nrow=100 exactly
[08] Assign column names
[09] Apply user overrides on column types
  After 0 type and 0 drop user overrides : AC557577777
[10] Allocate memory for the datatable
  Allocating 11 column slots (11 - 0 dropped) with 100 rows
[11] Read the data
  jumps=[0..1), chunk_size=1048576, total_size=1549842
fread.c:2373:26: runtime error: applying non-zero offset 16 to null pointer
    #0 0x129e1a5d0 in .omp_outlined.+0x4100 (data_table.so:arm64+0x8a5d0) (BuildId: b8d15fa651093043b749477677d067b732000000200000000100000000000d00)
    #1 0x129c7e138 in __kmp_invoke_microtask+0x98 (libomp.dylib:arm64+0x82138) (BuildId: c3eb6354ed623df891ae69a80b69e48f32000000200000000100000000000d00)
    #2 0x16b68425c  (<unknown module>)

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior fread.c:2373:26 in
fread.c:2373:26: runtime error: member access within address 0x000000000010 with insufficient space for an object of type 'lenOff'
0x000000000010: note: pointer points here
<memory cannot be printed>
    #0 0x129e1a5e8 in .omp_outlined.+0x4118 (data_table.so:arm64+0x8a5e8) (BuildId: b8d15fa651093043b749477677d067b732000000200000000100000000000d00)
    #1 0x129c7e138 in __kmp_invoke_microtask+0x98 (libomp.dylib:arm64+0x82138) (BuildId: c3eb6354ed623df891ae69a80b69e48f32000000200000000100000000000d00)
    #2 0x16b68425c  (<unknown module>)

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior fread.c:2373:26 in
fread.c:2373:38: runtime error: load of address 0x000000000010 with insufficient space for an object of type 'int32_t' (aka 'int')
0x000000000010: note: pointer points here
<memory cannot be printed>
    #0 0x129e1a5f8 in .omp_outlined.+0x4128 (data_table.so:arm64+0x8a5f8) (BuildId: b8d15fa651093043b749477677d067b732000000200000000100000000000d00)
    #1 0x129c7e138 in __kmp_invoke_microtask+0x98 (libomp.dylib:arm64+0x82138) (BuildId: c3eb6354ed623df891ae69a80b69e48f32000000200000000100000000000d00)
    #2 0x16b68425c  (<unknown module>)

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior fread.c:2373:38 in
```

I would recommend filing a bug report with the `data.table` team.

For reference, the code I used:

```
library(data.table)
nrow_read <- 1200000
nrow_skip <- 0
nrow_df <- nrow_read
swp_in <- data.table::fread(
  input = "~/Downloads/sweep_20230513_1100.log",
  header = F,
  integer64 = "numeric",
  skip = nrow_skip,
  nrow = nrow_read,
  verbose = TRUE
)
```

**Comment by chrisvwn** [comment]
Aha! I should have checked that. @jonvanausdeln Yes, it also crashes in R at the shell. This does seem to be an issue with data.table - thanks @kevinushey for confirming the bug and pointing out the R built with sanitizers link. Let me report it with them. Thanks!
